diff --git a/src/oxygenwindowmanager.cpp b/src/oxygenwindowmanager.cpp
index 4986384..c4ecedb 100644
--- a/src/oxygenwindowmanager.cpp
+++ b/src/oxygenwindowmanager.cpp
@@ -30,10 +30,6 @@
 #include "oxygenstyle.h"
 #include "config.h"
 
-#ifdef GDK_WINDOWING_X11
-#include <gdk/gdkx.h>
-#endif
-
 namespace Oxygen
 {
 
@@ -48,7 +44,8 @@ namespace Oxygen
         _widget( 0L ),
         _lastRejectedEvent( 0L ),
         _x(-1),
-        _y(-1)
+        _y(-1),
+        _time(0)
     {
         #if OXYGEN_DEBUG
         std::cerr << "Oxygen::WindowManager::WindowManager" << std::endl;
@@ -179,6 +176,7 @@ namespace Oxygen
             _widget = 0L;
             _x = -1;
             _y = -1;
+            _time = 0;
             _drag = false;
         }
 
@@ -390,51 +388,22 @@ namespace Oxygen
         if( distance < _dragDistance ) return false;
 
         // start drag from current position
-        return startDrag( widget, int(event->x_root), int(event->y_root) );
+        return startDrag( widget, int(event->x_root), int(event->y_root), event->time );
 
     }
 
     //_________________________________________________________________
-    bool WindowManager::startDrag( GtkWidget* widget, int x, int y )
+    bool WindowManager::startDrag( GtkWidget* widget, int x, int y, guint32 time )
     {
 
-        #ifdef GDK_WINDOWING_X11
         // create xevent and send.
-        XEvent xev;
         GtkWindow* topLevel = GTK_WINDOW( gtk_widget_get_toplevel( widget ) );
-        GdkWindow* window = gtk_widget_get_window( GTK_WIDGET( topLevel ) );
-        GdkDisplay* display = gtk_widget_get_display( GTK_WIDGET( topLevel ) );
-        GdkWindow* root = gdk_screen_get_root_window( gtk_window_get_screen( topLevel ) );
-
-        xev.xclient.type = ClientMessage;
-        xev.xclient.message_type = gdk_x11_get_xatom_by_name_for_display(display, "_NET_WM_MOVERESIZE");
-        xev.xclient.display = GDK_DISPLAY_XDISPLAY(display);
-        xev.xclient.window = GDK_WINDOW_XID(window);
-        xev.xclient.format = 32;
-        xev.xclient.data.l[0] = x;
-        xev.xclient.data.l[1] = y;
-        xev.xclient.data.l[2] = 8; // NET::Move
-        xev.xclient.data.l[3] = Button1;
-        xev.xclient.data.l[4] = 0;
-        XUngrabPointer(GDK_DISPLAY_XDISPLAY(display), CurrentTime);
-
-        XSendEvent(
-            GDK_DISPLAY_XDISPLAY(display),
-            GDK_WINDOW_XID(root),
-            False,
-            SubstructureRedirectMask | SubstructureNotifyMask,
-            &xev);
+        gtk_window_begin_move_drag( topLevel, 1, x, y, time );
 
         // force a release as some widgets miss it...
         finishDrag();
         return true;
 
-        #else
-
-        return false;
-
-        #endif
-
     }
 
     //_________________________________________________
@@ -445,6 +414,7 @@ namespace Oxygen
         _lastRejectedEvent = 0L;
         _x = -1;
         _y = -1;
+        _time = 0;
 
         // stop timer
         if( _timer.isRunning() ) _timer.stop();
@@ -452,10 +422,6 @@ namespace Oxygen
         if( _drag )
         {
 
-            #ifdef GDK_WINDOWING_X11
-            gdk_pointer_ungrab( CurrentTime );
-            #endif
-
             _drag = false;
             return true;
 
@@ -475,6 +441,7 @@ namespace Oxygen
             _widget = widget;
             _x = int(event->x_root);
             _y = int(event->y_root);
+            _time = event->time;
 
             // start timer
             if( _timer.isRunning() ) _timer.stop();
diff --git a/src/oxygenwindowmanager.h b/src/oxygenwindowmanager.h
index f8a774e..0687646 100644
--- a/src/oxygenwindowmanager.h
+++ b/src/oxygenwindowmanager.h
@@ -124,11 +124,11 @@ namespace Oxygen
         bool startDrag( GtkWidget*, GdkEventMotion* );
 
         //! start dragging widget
-        bool startDrag( GtkWidget*, int, int );
+        bool startDrag( GtkWidget*, int, int, guint32 );
 
         //! start dragging widget
         void startDrag( void )
-        { if( _drag && _widget ) startDrag( _widget, _x, _y ); }
+        { if( _drag && _widget ) startDrag( _widget, _x, _y, _time ); }
 
         //! finish dragging widget
         bool finishDrag( void );
@@ -249,6 +249,9 @@ namespace Oxygen
         int _x;
         int _y;
 
+        //! drag time
+        guint32 _time;
+
         //! widget typenames black-list
         std::vector<std::string> _blackList;
 
