diff --git a/src/oxygentileset.cpp b/src/oxygentileset.cpp
index 371e123..5d4def2 100644
--- a/src/oxygentileset.cpp
+++ b/src/oxygentileset.cpp
@@ -212,9 +212,9 @@ namespace Oxygen
 
             } else {
 
-                #if CAIRO_VERSION >= CAIRO_VERSION_ENCODE(1, 10, 0)
-                Cairo::Surface tile( cairo_surface_create_for_rectangle( source, sx, sy, sw, sh ) );
-                #else
+//                 #if CAIRO_VERSION >= CAIRO_VERSION_ENCODE(1, 10, 0)
+//                 Cairo::Surface tile( cairo_surface_create_for_rectangle( source, sx, sy, sw, sh ) );
+//                 #else
                 Cairo::Surface tile( cairo_surface_create_similar( source, CAIRO_CONTENT_COLOR_ALPHA, sw, sh ) );
                 {
                     Cairo::Context local( tile );
@@ -222,7 +222,7 @@ namespace Oxygen
                     cairo_rectangle( local, 0, 0, sw, sh );
                     cairo_fill( local );
                 }
-                #endif
+//                 #endif
 
                 cairo_set_source_surface( context, tile, 0, 0 );
                 cairo_pattern_set_extend( cairo_get_source( context ), CAIRO_EXTEND_REPEAT );
